From 852b5b38fe2e1c9ca1cdd4ca3ed46c7c578842ae Mon Sep 17 00:00:00 2001
Message-Id: <852b5b38fe2e1c9ca1cdd4ca3ed46c7c578842ae.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 13 Aug 2014 13:54:41 +0530
Subject: [PATCH 251/312] FOR_UPSTREAM [VPG]: drm/i915: Set min freq to
 efficient freq for better performance at Vmin

On CHT, By setting the freq starting point to efficient which is max
freq at Vmin, By doing so, we can get better performance in video
decoding scenarios, where the workload might not generate gfx business
to trigger an interrupt.

Issue: GMIN-3330
Change-Id: Ied85521313f7dde93683793c4b16d6c046194a4f
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 086297c..0495d91 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -4644,7 +4644,7 @@ static void cherryview_init_gt_powersave(struct drm_device *dev)
 		dev_priv->rps.max_freq_softlimit = dev_priv->rps.max_freq;
 
 	if (dev_priv->rps.min_freq_softlimit == 0)
-		dev_priv->rps.min_freq_softlimit = dev_priv->rps.min_freq;
+		dev_priv->rps.min_freq_softlimit = dev_priv->rps.efficient_freq;
 
 	mutex_unlock(&dev_priv->rps.hw_lock);
 }
-- 
1.7.9.5

